src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets

```diff
@@ -70,6 +70,7 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <Target Name="GenerateBuildDependencyFile"
           DependsOnTargets="_DefaultMicrosoftNETPlatformLibrary"
+          BeforeTargets="_CheckForCompileOutputs"
```

Comment:
Wouldn't this cause GenerateBuildDependencyFile to run on Clean? That's not right. We want it to only run on build and log the write then.